library/common/jni/jni_interface.cc

```diff
@@ -496,15 +474,7 @@ jvm_http_filter_on_resume(const char* method, envoy_headers* headers, envoy_data
   }
   jbyteArray j_in_data = NULL;
   if (data) {
-    j_in_data = env->NewByteArray(data->length);
-    // TODO: check if copied via isCopy.
-    // TODO: check for NULL.
-    // https://github.com/lyft/envoy-mobile/issues/758
-    void* critical_data = env->GetPrimitiveArrayCritical(j_in_data, nullptr);
-    memcpy(critical_data, data->bytes, data->length);
-    // Here '0' (for which there is no named constant) indicates we want to commit the changes back
-    // to the JVM and free the c array, where applicable.
-    env->ReleasePrimitiveArrayCritical(j_in_data, critical_data, 0);
+    j_in_data = native_data_to_array(env, *data);
```

Comment:
There is no call to `env->DeleteLocalRef(j_in_data)` below. 

Should there be a:

```c
if (data) {
  env->DeleteLocalRef(j_in_data);
}
```

block below?

Reply:
I think we do need that deletion